pkg/cableengine/ipsec/strongswan.go

```diff
@@ -291,15 +315,71 @@ func (i *strongSwan) loadSharedKey(endpoint types.SubmarinerEndpoint, client *go
 	return nil
 }
 
-func runCharon(debug bool, logFile string) error {
+// charonConfTemplate defines the configuration for strongswan IKE keying daemon,
+// * port and port_nat_t define the IKE and IKE NATT UDP ports
+// * make_before_break ensures dataplane connectivity while re-authenticating endpoints (check this)
+// TODO : check what * ignore_acquire_ts means
+const charonConfTemplate = `
+	charon {
+		port = {{.ipSecIKEPort}}
+		port_nat_t = {{.ipSecNATTPort}}
+		make_before_break = yes
+		ignore_acquire_ts = yes
+	}
+`
+
+func (i *strongSwan) writeCharonConfig(path string) error {
+	err := os.Remove(path)
+	if err != nil {
+		klog.Warningf("Error deleting %s: %s", path, err)
+	}
+
+	f, err := os.Create(path)
+
+	if err != nil {
+		return fmt.Errorf("Error creating %s: %s", path, err)
+	}
+	if err = i.renderCharonConfigTemplate(f); err != nil {
+		return err
+	}
+
+	if err = f.Close(); err != nil {
+		return fmt.Errorf("Unable to close %s: %s", path, err)
+	}
+
+	return nil
+}
+
+func (i *strongSwan) renderCharonConfigTemplate(f io.Writer) error {
+	t, err := template.New("charon.conf").Parse(charonConfTemplate)
+	if err != nil {
+		return fmt.Errorf("Error creating template for charon.conf: %s", err)
+	}
+
+	err = t.Execute(f, map[string]string{
+		"ipSecIKEPort":  i.ipSecIKEPort,
+		"ipSecNATTPort": i.ipSecNATTPort})
+
+	if err != nil {
+		return fmt.Errorf("Error rendering charon config file: %s", err)
+	}
+	return nil
+}
+
+func (i *strongSwan) runCharon() error {
+
+	if err := i.writeCharonConfig(strongswanCharonConfigFilePath); err != nil {
+		klog.Fatalf("Error writing strongswan charon config: %s", err)
```

Comment:
Can we propagate the error here as well?

Reply:
Done. I missed that one.